ts/modules/marker-clusters.ts

```diff
@@ -0,0 +1,1948 @@
+
+import './../parts/Series.js';
+import './../parts/Axis.js';
+import './../parts/SvgRenderer.js';
+
+var Series = H.Series,
+    Scatter = H.seriesTypes.scatter,
+    Point = H.Point,
+    SvgRenderer = H.SVGRenderer,
+    addEvent = H.addEvent,
+    merge = H.merge,
+    defined = U.defined,
+    isArray = U.isArray,
+    isObject = U.isObject,
+    isFunction = H.isFunction,
+    isNumber = U.isNumber,
+    pick = H.pick,
+    relativeLength = H.relativeLength,
+    error = H.error,
+    objectEach = U.objectEach,
+    baseGeneratePoints = Series.prototype.generatePoints,
+    clusterDefaultOptions;
+
+/**
+ * Options for marker clusters, the concept of sampling the data
+ * values into larger blocks in order to ease readability and
+ * increase performance of the JavaScript charts.
+ *
+ * Note: marker clusters module is not working with `boost`
+ * and `draggable-points` modules.
+ *
+ * The marker clusters feature requires the marker-clusters.js
+ * file to be loaded, found in the modules directory of the download
+ * package, or online at [code.highcharts.com/modules/marker-clusters.js
+ * ](code.highcharts.com/modules/marker-clusters.js).
+ *
+ * @sample maps/marker-clusters/europe
+ *         Maps marker clusters
+ * @sample highcharts/marker-clusters/basic
+ *         Scatter marker clusters
+ * @sample maps/marker-clusters/optimized-kmeans
+ *         Marker clusters with colorAxis
+ *
+ * @product      highcharts highmaps
+ * @since        next
+ * @optionparent plotOptions.scatter.marker.cluster
+ */
+clusterDefaultOptions = {
+    /**
+     * Whether to enable the marker-clusters module.
+     *
+     * @sample maps/marker-clusters/basic
+     *         Maps marker clusters
+     * @sample highcharts/marker-clusters/basic
+     *         Scatter marker clusters
+     *
+     * @type      {boolean}
+     * @default   false
+     */
+    enabled: false,
+    /**
+     * When set to `false` prevent cluster overlapping - this option
+     * works only when `layoutAlgorithm.type = "grid"`.
+     *
+     * @sample highcharts/marker-clusters/grid
+     *         Prevent overlapping
+     *
+     * @type      {boolean}
+     * @default   true
+     */
+    allowOverlap: true,
+    /**
+     * Zoom the plot area to the cluster points range when cluster is clicked.
+     *
+     * @type      {boolean}
+     * @default   true
+     */
+    drillToCluster: true,
+    /**
+     * The minimum amount of points to be combined into a cluster.
+     * This value has to be greater or equal to 2.
+     *
+     * @sample highcharts/marker-clusters/basic
+     *         At least three points in the cluster
+     *
+     * @type      {number}
+     * @default   2
+     */
+    minimumClusterSize: 2,
+    /**
+     * Options for layout algorithm. Inside there
+     * are options to change the type of the algorithm, gridSize,
+     * distance or iterations.
+     */
+    layoutAlgorithm: {
+        /* eslint-disable max-len */
```

Comment:
If this is needed here, we should instead add `@optionparent` to the lines, that get ignored by eslint. What do you think about this @TorsteinHonsi ?